### PR TITLE
Decouple gitpod web extension

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

Decouple gitpod web extension

## How to test

1. Open workspace and check gitpod-web extension is activated

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [x] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/ide/code/gitpod-web/BUILD.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+# Licensed under the GNU Affero General Public License (AGPL).
+# See License.AGPL.txt in the project root for license information.
+
+packages:
```
Comment (akosyakov) [inline components/ide/code/gitpod-web/BUILD.yaml:5]: Do we need this component? I hoped we can remove eventually all code related component out of this repo and only configure installer in code.

Can we publish a ready image already?

**Comment by akosyakov** [comment]
We need to think how we are going to rollout. Could there be some issues with race conditions, ie some start a workpace but ide service does not resolve yet new images?

File: components/ide/code/gitpod-web/BUILD.yaml (hunk 1 above)
Comment (jeanp413) [reply components/ide/code/gitpod-web/BUILD.yaml:5]: @mustard-mh will check tomorrow morning about publishing to gpc registry

**Comment by mustard-mh** [comment]
Addressed, and tested, going to add new iam in GCP

image

File: components/ide/code/gitpod-web/BUILD.yaml (hunk 1 above)
Comment (mustard-mh) [reply components/ide/code/gitpod-web/BUILD.yaml]: https://github.com/gitpod-io/gitpod/pull/16073#issuecomment-1410099081

**Comment by mustard-mh** [comment]
> We need to think how we are going to rollout. Could there be some issues with race conditions, ie some start a workpace but ide service does not resolve yet new images?

@akosyakov  We have no race conditions, as this value is constant for all commits, `ide-service` only will return with the previous value of the latest value, and they are both correct (we will need to write compatibility code in `gitpod-io/gitpod-code` repo.

**Comment by akosyakov** [comment]
@mustard-mh for latest should be alright too?

**Comment by mustard-mh** [comment]
> for latest should be alright too?

@akosyakov  It will be a layer of image for both latest and stable browser code, so it (`gitpod-web`) will always be the one we configured in `constants.go`

**Review by mustard-mh — APPROVED** [review]

**Comment by jeanp413** [comment]
There's errors in console, I remember seeing these before it's related to compilation I think, looking into it
```
cannot maintain connection to supervisor Error: 13 INTERNAL: Request message serialization failure: Expected argument of type supervisor.PortsStatusRequest
```

**Comment by mustard-mh** [comment]
https://github.com/gitpod-io/gitpod/pull/16073#issuecomment-1410946708

> There's errors in console, I remember seeing these before it's related to compilation I think, looking into it
> 
> ```
> cannot maintain connection to supervisor Error: 13 INTERNAL: Request message serialization failure: Expected argument of type supervisor.PortsStatusRequest
> ```

Browser code in preview env is out of date, cherry-pick commits in `gp-code/main` may fix it (also include a fixing for `loggedInUserID`) cc @jeanp413

**Comment by jeanp413** [comment]
yeah I missed the last commit in gitpod-shared fixing the bug in `loggedInUserID` also there was a bug in vscode compiling web release so had to rebase again :facepalm:

**Comment by jeanp413** [comment]
To fix for the serialization error I had to refactor a little bit gitpod-shared

**Pull request merged**